ingestion/src/main/java/feast/ingestion/transform/fn/ValidateFeatureRowDoFn.java

```diff
@@ -81,6 +82,7 @@ public void processElement(ProcessContext context) {
             break;
           }
         }
+        fields.add(field);
```

Comment:
Will a collision be happening here? And if so, how will it be handled?

Reply:
Would it make sense to detect collision at the name level?

Reply:
If there is a collision the 2nd element will not be written. I'm not sure if i want to commit to detecting collision at the name level at this point because it could result in data loss (that would go uncaught by the user)

Reply:
Ok, so if we have two fields with the same name, what happens to the data when it is written to the store? (in the most pathological case)